src/utils/assets.js

```diff
@@ -63,59 +76,69 @@ export function getRate(rates: Rates = {}, token: string, fiatCurrency: string):
   }
 
   return ethToFiat * tokenToETH;
-}
+};
 
-export function calculateMaxAmount(token: string, balance: number | string, txFeeInWei: BigNumber): number {
+export const calculateMaxAmount = (token: string, balance: number | string, txFeeInWei: BigNumber): number => {
   if (typeof balance !== 'string') {
     balance = balance.toString();
   }
+
   if (token !== ETH) {
     return +balance;
   }
+
   const maxAmount = utils.parseUnits(balance, 'ether').sub(txFeeInWei);
   if (maxAmount.lt(0)) return 0;
+
   return new BigNumber(utils.formatEther(maxAmount)).toNumber();
-}
+};
 
-export function checkIfEnoughForFee(balances: Balances, txFeeInWei: BigNumber): boolean {
+export const checkIfEnoughForFee = (balances: Balances, txFeeInWei: BigNumber): boolean => {
   if (!balances[ETH]) return false;
   const ethBalance = getBalance(balances, ETH);
   const balanceInWei = utils.parseUnits(ethBalance.toString(), 'ether');
   return balanceInWei.gte(txFeeInWei);
-}
-
-export function calculatePortfolioBalance(assets: Assets, rates: Rates, balances: Object) {
-  // CLEANUP REQUIRED
-  return Object
-    .keys(assets)
-    .map(key => assets[key])
-    .map(({ symbol }) => {
-      const assetRates = rates[symbol] || {};
-      const balance = getBalance(balances, symbol);
-      const assetFiatBalance = Object
-        .keys(assetRates)
-        .map(key => ({
-          currency: key,
-          total: assetRates[key] * (balance || 0),
-        }));
-      return assetFiatBalance;
-    })
-    .reduce((memo, item) => {
-      return memo.concat(item);
-    }, [])
-    .reduce((memo, item) => {
-      memo[item.currency] = (memo[item.currency] || 0) + item.total;
-      return memo;
-    }, { GBP: 0 });
-}
-
-export function getPPNTokenAddress(token: string, assets: Assets): ?string {
+};
+
+const balanceInEth = (balances: Balances, rates: Rates): number => {
+  const balanceValues: Balance[] = Object.keys(balances).map(key => balances[key]);
+
+  return balanceValues.reduce((total, item) => {
+    const balance = +item.balance;
+    const assetRates = rates[item.symbol];
+
+    if (!assetRates || balance === 0) {
+      return total;
+    }
+
+    return total + (assetRates[ETH] * balance);
```

Comment:
What would happen if there won't be `assetRates[ETH]` set?

Reply:
Pain would happen, good catch!

Reply:
Updated.